File: crates/cxx-qt-lib/src/core/qtranslator.cpp
#include "cxx-qt-lib/qtranslator.h"
#include <QFile>
#include <QIODevice>
#include <QDebug>
#include <QDir>
#include <QStringList>
namespace rust {
namespace cxxqtlib1 {

::std::unique_ptr<QTranslator>
qtranslatorNew()
{
    auto translator = ::std::make_unique<QTranslator>();
    QCoreApplication::installTranslator(translator.get());
    return translator;
}

bool
loadTranslation(std::unique_ptr<QTranslator> translator, const QString& qmFilePath)
{
    if (!translator->load(qmFilePath)) {
        qDebug() << "Failed to load translation file:" << qmFilePath;
        return false;
    }
    
    // Make sure the translator is installed
    if (!QCoreApplication::installTranslator(translator.get())) {
        qDebug() << "Failed to install translator";
        return false;
    }
    
    return true;
}
}
}

File: crates/cxx-qt-lib/src/core/qtranslator.rs
#[cxx::bridge]
mod ffi {
    unsafe extern "C++" {
        type QString = crate::QString;
        include!("cxx-qt-lib/common.h");
        include!("cxx-qt-lib/qtranslator.h");
        type QTranslator;
    }

    #[namespace = "rust::cxxqtlib1"]
    unsafe extern "C++" {
        #[doc(hidden)]
        #[rust_name = "qtranslator_new"]
        fn qtranslatorNew() -> UniquePtr<QTranslator>;

        #[doc(hidden)]
        #[rust_name = "qtranslator_load_translation"]
        fn loadTranslation(
            ptr: UniquePtr<QTranslator>,
            qmFilePath: &QString) -> bool;
    }
}

pub use ffi::QTranslator;

impl QTranslator {
    pub fn load_translation(ptr: cxx::UniquePtr<QTranslator>, qm_file_path: &str) -> bool {
        // 将 &str 转换为 &QString
        let qm_file_path = ffi::QString::from(qm_file_path);
        ffi::qtranslator_load_translation(ptr, &qm_file_path)
    }

    pub fn new() -> cxx::UniquePtr<Self> {
        ffi::qtranslator_new()
    }
}
